### ISSUE TITLE
Different convolutional padding per channel

### ISSUE THREAD

**Issue by squared9** [tensorflow/tensorflow#15883]
Hello,

would it be possible to specify different convolutional padding for each channel? My use case is using 3D convolutions on input, where the first dimension is temporal and the next 2 dimensions are 2D images. I'd like to have "SAME" padding on temporal dimension and "VALID" padding on convolutional dimensions.

The model in question is for estimating speed of a vehicle from on-board camera where a sequence of images of fixed length (e.g. 10 frames) is sent to a 3D convolutional network and it makes sense to avoid unnecessary clipping of temporal dimension where it's perfectly fine for convolutional part.

Would this be possible to add to TensorFlow?

Thank you!

Have I written custom code: No
OS Platform and Distribution: Linux Mint 18.2
TensorFlow installed from: pip
TensorFlow version: 1.3.0
Bazel version: N/A
CUDA/cuDNN version: 9.0
GPU model and memory: GTX970 4GB
Exact command to reproduce: tf.nn.conv2d's "padding" parameter uses a single approach for all convolutional dimensions

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by poxvoculi** [comment]
@martinwicke Whose domain does this feature request fall into?

**Comment by martinwicke** [comment]
@fchollet @sherrym for additional opinions.

This may not be worth the added API complexity. Have you tried using "same" for all dimensions and clipping manually afterwards? There's a performance cost to this, but it's probably negligible compared to the cost of the convolution itself.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 44 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
We are closing this issue for now due to lack of activity. Please comment if this is still an issue for you. Thanks!
